src/common/redis_error.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <map>
+
+namespace redis {
+
+enum class ErrorKind {
```

Comment:
Using `RedisError` ?

Reply:
ErrorKind => RedisError? If yes, perhaps `Error` is better name since it was inside `redis` namespace now.

Reply:
Yeah, I think the suffix of `Kind` is a bit redundant :)

Reply:
But it's an error kind instead of error indeed, I prefer keeping this suffix.

Reply:
OK, got it.